src/vanilla.ts

```diff
@@ -5,7 +5,7 @@ type SetStateInternal<T> = {
   ): void
 }['_']
 
-export interface StoreApi<T extends object = object> {
+export interface StoreApi<T = any> {
```

Comment:
Can this be `= unknown`?

Reply:
It should be just `StoreApi<T>`, the default type parameter is not needed.

Reply:
We use it for `extends StoreApi` in other places. Should `T` be `any` or `unknown` there?

Reply:
`unknown`. Also ideally one should write `StoreApi<unknown>` and maybe we can make the type parameter required in v5.

Reply:
Thanks. It was my wrong change in #1122 then. I consider this as bug fix.

Reply:
Ah I see so it was accidentally introduced it in #1122, and it's not that I missed removing it in #725 haha.

> I consider this as bug fix

Yes, but it'd still be a breaking change for v4.0.0 users (not v3.x.x), so would you wait for v5 or publish it in v4.0.1?

Reply:
If people use it like `<StoreApi>`, it's breaking, but it's totally unexpected and "wrong" usage per se. So, I think it's good to "fix" this in v4.

Reply:
It still would technically violate semver, perhaps one could argue it doesn't violate on basis that users are not supposed to rely on undocumented behavior ie "wrong usage" as you said, nevertheless it's worth violating if at all it's a violation.